.github/funcs.sh

```diff
@@ -25,9 +25,45 @@ get_minor() {
 }
 
 get_patch() {
-    echo "${1}" | cut -d'.' -f3 | cut -d'-' -f1
+    if [[ ${1} == *"-"* ]]; then
+        echo "${1}" | cut -d'.' -f3 | cut -d'-' -f1
+    else
+        echo "${1}" | cut -d'.' -f3
+    fi
 }
 
 get_maturity() {
-    echo "${1}" | cut -d'.' -f3 | cut -d'-' -f2
+    if [[ ${1} == *"-"* ]]; then
+        echo "${1}" | cut -d'.' -f3 | cut -d'-' -f2
+    else
+        echo ""
+    fi
+}
+
+get_docker_tags() { 
+    # Arguments:
+    #  - Name reference for the array 
+    #  - Name of the docker image
+    #  - The semver we are using to create the tags
+
+    local -n arr=$1             # use nameref to create values
+    name=$2
+    semver=$3
+
+    major=$(get_major "${semver}")
+    minor=$(get_minor "${semver}")
+    patch=$(get_patch "${semver}")
+    maturity=$(get_maturity "${semver}")
+
+    echo "Detected Version: ${major} . ${minor} . ${patch} [${maturity}]"
```

Comment:
Nit: Might be nice to move the echo into the if/else to avoid an empty `[]` when no maturity is defined